src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,347 @@
+import Save from "@vertigis/web/ui/icons/Save";
+import Undo from "@vertigis/web/ui/icons/Undo";
+import Cancel from "@vertigis/web/ui/icons/Cancel";
+
+import * as locale from "@vertigis/arcgis-extensions/locale";
+import * as numberUtils from "@vertigis/arcgis-extensions/utilities/number";
+
+type FooterRow = Record<string, any>;
+type RowData = Record<string, any>;
+type SettableBoxProps = Pick<BoxProps, "maxHeight" | "maxWidth">;
+type SettableTableProps = Pick<TableProps, "size" | "stickyHeader">;
+
+interface Column {
+    name: string;
+    alias?: string;
+    align?: TableCellProps["align"];
+    columnCalculation?: "sum" | "average" | ((values: any) => any);
+    editable?: boolean;
+    type?: "text" | "number";
+    step?: string;
+    numberFormat?: numberUtils.FormatOptions;
+    footerFormat?: numberUtils.FormatOptions;
+}
+
+interface EditableTableElementProps
+    extends FormElementProps<RowData[]>,
+    SettableBoxProps,
+    SettableTableProps {
+    cols: Column[];
+    rows: RowData[];
+    footerLabel?: string;
+    onMouseEnter?: (row: RowData) => void;
+    onMouseLeave?: (row: RowData) => void;
+    onRowEdit?: (row: RowData) => RowData;
+}
+
+//Mimic MUI header style for footer
+const FooterCellStyleOverrides: TableCellProps["sx"] = {
+    fontFamily: "var(--defaultFont)",
+    backgroundColor: "var(--secondaryBackground)",
+    fontWeight: "bold",
+    color: "var(--secondaryForeground)",
+    fontSize: "inherit",
+};
+
+//Mimic MUI header style for footer
+const TableCellStyleOverrides: TableCellProps["sx"] = {
+    fontFamily: "var(--defaultFont)",
+    fontSize: "inherit",
+};
+
+const InputStyleOverrides: InputProps["sx"] = {
+    ".MuiInput-input": {
+        borderRadius: "4px !important",
+        padding: "5px 8px !important",
+        border: "0px !important",
+        backgroundColor: "inherit",
+        width: "100%",
+        height: "auto",
+    },
+};
+
+const IconButtonStyleOverrides: IconButtonProps["sx"] = {
+    // Hack required to override .gcx-forms.defaults
+    borderRadius: "50% !important",
+    // Hack required to match disabled list item content
+    "&.Mui-disabled": {
+        opacity: 0.38,
+    },
+};
+const calculateFooterRow = (cols: Column[], rows: RowData[]): FooterRow => {
+    const totalCols = cols.filter((x) => x.columnCalculation);
+    const total: FooterRow = {};
+    for (const col of totalCols) {
+        if (col.columnCalculation) {
+            const values = rows.map((x) => x[col.name]);
+            if (col.columnCalculation === "sum") {
+                total[col.name] = calculateSum(values);
+            } else if (col.columnCalculation === "average") {
+                total[col.name] = calculateAverage(values);
+            } else {
+                total[col.name] = col.columnCalculation(values);
+            }
+        }
+    }
+    return total;
+};
+
+const setLocale = (formatOptions: numberUtils.FormatOptions): numberUtils.FormatOptions => {
```

Comment:
We should not need to do this. It uses the current locale of the app by default. That is what we want.

Reply:
Fixed